### PR TITLE
chore: further automate release process (#462)

### LINKED ISSUE #462: COMMON: Automate sementic version determination
Based on commit messages it's possible to determine if a change to a chart is major, minor or patch. The main person who should know which version to bump is the developer. So he could mark his feature-commit (features should be squashed) with the needed information.

How to use semantic versioning to determine the version increment, e.g.: https://python-semantic-release.readthedocs.io/en/latest/commit-parsing.html

At the moment we use [this](https://github.com/intershop/helm-charts/actions/workflows/release-branch.yml) action to build a release branch from a develop branch. With an input like "icm-as:minor" the action could be started. But this input could be also gained from the information whitin the commit messages of the branch differences. E.g. one commit message contains "BREAKING CHANGE" and addresses a file in the icm chart then "icm:major" should be used for the release process.

### PR DESCRIPTION
**Pull request by khauser**
## PR Type

- [x] Build-related changes

## What Is the Current Behavior?

Currently charts and the semantic version type are needed for a release [release](https://github.com/intershop/helm-charts/actions/workflows/release-branch.yml) pipeline (e.g. `icm-as:minor`). This will be changed by using an automatic approach where commits and theire messages deliver the needed information.

Issue Number: Closes #462

## What Is the New Behavior?

Chart changes could be released by just starting the [github action](https://github.com/intershop/helm-charts/actions/workflows/release-branch.yml) and selecting the branch to release. Afterwards a PullReqest will be made with the changes.

## Does this PR Introduce a Breaking Change?

- [x] No

## Other Information
* the pullrequest also switches from [bump2version](https://github.com/c4urself/bump2version) to [bump-my-version](https://github.com/callowayproject/bump-my-version)

### CODE REVIEW

File: .github/template/bump-version/action.yml [hunk 1]
```diff
@@ -14,5 +14,9 @@ runs:
   - name: Bump Version
     shell: bash
     run: |
-      pip install --no-cache-dir bump2version
+      if [[ -z "${{inputs.changes}}" ]]; then
+        echo "Missing parameter inputs.changes" 1>&2
+        exit 1
+      fi
+      pip install --no-cache-dir bump-my-version
```
Comment (cortlepp-intershop) [inline .github/template/bump-version/action.yml:21]: I think that we should specify a version here (for reproducibility).

File: .github/workflows/release-branch.yml [hunk 2]
```diff
@@ -2,18 +2,128 @@ name: Create Release Branch
...
+        else
+          echo "Only development branches could be released" 1>&2
+          exit 1
+        fi
+
+    - name: Get list of changed charts
+      id: list-changed-charts
+      uses: tj-actions/changed-files@v40.1.1
+      with:
+        dir_names_max_depth: 2
```
Comment (cortlepp-intershop) [inline .github/workflows/release-branch.yml:41]: Why is this limitation introduced?

**Comment by cortlepp-intershop** [comment]
We should also update the PR template to reflect the new guidelines and that following them is mandatory. Maybe we could even require it in some way by a action, but that may be too hard to do (but maybe this already exists).

File: .github/workflows/release-branch.yml (hunk 2 above)
Comment (khauser) [reply .github/workflows/release-branch.yml:41]: Yes you are right. This only need is directories are delivered.

File: .github/template/bump-version/action.yml (hunk 1 above)
Comment (khauser) [reply .github/template/bump-version/action.yml]: Sure

**Comment by khauser** [comment]
I also updated the PR template and added a link to the [wiki](https://github.com/intershop/helm-charts/wiki/Release-Process).

**Review by cortlepp-intershop — APPROVED** [review]

**Pull request merged**